pandas/tseries/index.py

```diff
@@ -1389,14 +1389,16 @@ def time(self):
         """
         # can't call self.map() which tries to treat func as ufunc
         # and causes recursion warnings on python 2.6
-        return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values, lambda x: x.time()))
+        return self._maybe_mask_results(_algos.arrmap_object(self.asobject.values,
```

Comment:
so its better to actually define some of these (e.g. as we did for ``weekday`` to directly return ``np.nan``), so ``.date()``/``.time()`` should simply return ``np.nan``